code-scanning/parasoft-dottest.yml

```diff
@@ -0,0 +1,107 @@
+# This workflow uses actions that are not certified by GitHub.
+# They are provided by a third-party and are governed by
+# separate terms of service, privacy policy, and support
+# documentation.
+
+# This workflow runs Parasoft dotTEST to analyze code
+# and display results with Github code scanning alerts.
+# Parasoft dotTEST is a testing tool that provides code analysis techniques
+# to improve code quality and ensure compliance with industry standards.
+# See https://github.com/parasoft/run-dottest-action for more information.
+
+name: Parasoft dotTEST Code Analysis
+
+on:
+  push:
+    branches: [ $default-branch, $protected-branches ]
+  pull_request:
+    # The branches below must be a subset of the branches above.
+    branches: [ $default-branch ]
+  schedule:
+    - cron: $cron-weekly
+
+# A workflow run is made up of one or more jobs that can run sequentially or in parallel.
+jobs:
+  # This workflow is made up of one job "run-dottest-action".
+  run-dottest-action:
+    # Specifies the name of the job.
+    name: Run code analysis with dotTEST
+    # Specifies the type of runner that the job will run on.
+    runs-on: self-hosted
```

Comment:
```suggestion
    runs-on: ubuntu-20.04
```

Reply:
Current licensing mechanism does not allow to run the analysis on GitHub hosted runners. Is it a blocker? We are working to resolve this. I expect this to be working in the second half of the year.

Reply:
I'm not sure this is the user experience we want to promote in this starter workflow. @josepalafox will follow up with you by email 🙇.

Reply:
@imjohnbo @josepalafox, based on e-mail conversation, is it still a blocker or can we proceed and accept this approach?